checkbox-support/checkbox_support/dbus/monitor_config.py

```diff
@@ -0,0 +1,132 @@
+# Copyright 2024 Canonical Ltd.
+# Written by:
+#   Paolo Gentili <paolo.gentili@canonical.com>
+#
+# This is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3,
+# as published by the Free Software Foundation.
+#
+# This file is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this file.  If not, see <http://www.gnu.org/licenses/>.
+"""
+This modules includes a utility to get display information and set
+a new logical monitor configuration via DBus and Mutter.
+
+Original script that inspired this class:
+- https://gitlab.gnome.org/GNOME/mutter/-/blob/main/tools/get-state.py
+"""
+
+from typing import Dict, List, Tuple
+from gi.repository import GLib, Gio
+
+
+class MonitorConfigDBus:
+    """Get and modify the current Monitor configuration via DBus."""
+
+    NAME = "org.gnome.Mutter.DisplayConfig"
+    INTERFACE = "org.gnome.Mutter.DisplayConfig"
+    OBJECT_PATH = "/org/gnome/Mutter/DisplayConfig"
+
+    def __init__(self):
+        self._proxy = Gio.DBusProxy.new_for_bus_sync(
+            bus_type=Gio.BusType.SESSION,
+            flags=Gio.DBusProxyFlags.NONE,
+            info=None,
+            name=self.NAME,
+            object_path=self.OBJECT_PATH,
+            interface_name=self.INTERFACE,
+            cancellable=None,
+        )
+
+    def get_current_resolutions(self) -> Dict[str, str]:
+        """Get current active resolutions for each monitor."""
+
+        state = self._get_current_state()
+        return {
```

Comment:
Applies to the whole file, namely, when you use an item it would be useful to do so by name. I would use a namedtuple and "waste" one operation to caste the state arr to it (`MyType(*state)`) else this is very hard to read imo.

Alternatively you can use named variables/enums as indeces, but it is not as good/easy to read imo